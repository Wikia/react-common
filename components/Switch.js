'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var PropTypes = _interopDefault(require('prop-types'));
var classNames = _interopDefault(require('classnames'));

/**
 * **NOTE**: This component should only be used with `<Switch.Item>`,
 * see `<Switch>` above for a complete example.
 */

var SwitchWrapper = function SwitchWrapper(_ref) {
  var children = _ref.children,
      className = _ref.className;
  var allClassNames = classNames('switch-wrapper', className);
  return React.createElement("div", {
    className: allClassNames
  }, children);
};

SwitchWrapper.propTypes = {
  /** @ignore */
  children: PropTypes.node.isRequired,

  /** Additional class name */
  className: PropTypes.string
};
SwitchWrapper.defaultProps = {
  className: undefined
};

/**
 * **NOTE**: This component should only be used with `<Switch.Wrapper>`,
 * see `<Switch>` above for a complete example.
 */

var SwitchItem = function SwitchItem(_ref) {
  var active = _ref.active,
      children = _ref.children,
      className = _ref.className,
      onClick = _ref.onClick;
  var allClassNames = classNames('switch-item', className, {
    'switch-item__active': active
  });
  return React.createElement("button", {
    type: "button",
    className: allClassNames,
    onClick: onClick
  }, children);
};

SwitchItem.propTypes = {
  /** is the current item active */
  active: PropTypes.bool,

  /** @ignore */
  children: PropTypes.node.isRequired,

  /** Additional class name */
  className: PropTypes.string,

  /** Function fired when clicking on the element */
  onClick: PropTypes.func
};
SwitchItem.defaultProps = {
  active: false,
  className: undefined,
  onClick: undefined
};

/**
 * `<Switch>` component is built from two sub-components,
 * `<Switch.Wrapper>` and `<Switch.Item>`.
 *
 * @component
 */

var index = {
  Wrapper: SwitchWrapper,
  Item: SwitchItem
};

module.exports = index;
