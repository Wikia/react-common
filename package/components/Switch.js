'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var PropTypes = _interopDefault(require('prop-types'));

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var classnames = createCommonjsModule(function (module) {
/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/
/* global define */

(function () {

	var hasOwn = {}.hasOwnProperty;

	function classNames () {
		var classes = [];

		for (var i = 0; i < arguments.length; i++) {
			var arg = arguments[i];
			if (!arg) continue;

			var argType = typeof arg;

			if (argType === 'string' || argType === 'number') {
				classes.push(arg);
			} else if (Array.isArray(arg) && arg.length) {
				var inner = classNames.apply(null, arg);
				if (inner) {
					classes.push(inner);
				}
			} else if (argType === 'object') {
				for (var key in arg) {
					if (hasOwn.call(arg, key) && arg[key]) {
						classes.push(key);
					}
				}
			}
		}

		return classes.join(' ');
	}

	if (module.exports) {
		classNames.default = classNames;
		module.exports = classNames;
	} else {
		window.classNames = classNames;
	}
}());
});

/**
 * **NOTE**: This component should only be used with `<Switch.Item>`,
 * see `<Switch>` above for a complete example.
 */

var SwitchWrapper = function SwitchWrapper(_ref) {
  var children = _ref.children,
      className = _ref.className;
  var allClassNames = classnames('switch-wrapper', className);
  return React.createElement("div", {
    className: allClassNames
  }, children);
};

SwitchWrapper.propTypes = {
  /** @ignore */
  children: PropTypes.node.isRequired,

  /** Additional class name */
  className: PropTypes.string
};
SwitchWrapper.defaultProps = {
  className: undefined
};

/**
 * **NOTE**: This component should only be used with `<Switch.Wrapper>`,
 * see `<Switch>` above for a complete example.
 */

var SwitchItem = function SwitchItem(_ref) {
  var active = _ref.active,
      children = _ref.children,
      className = _ref.className,
      onClick = _ref.onClick;
  var allClassNames = classnames('switch-item', className, {
    'switch-item__active': active
  });
  return React.createElement("button", {
    type: "button",
    className: allClassNames,
    onClick: onClick
  }, children);
};

SwitchItem.propTypes = {
  /** is the current item active */
  active: PropTypes.bool,

  /** @ignore */
  children: PropTypes.node.isRequired,

  /** Additional class name */
  className: PropTypes.string,

  /** Function fired when clicking on the element */
  onClick: PropTypes.func
};
SwitchItem.defaultProps = {
  active: false,
  className: undefined,
  onClick: undefined
};

/**
 * `<Switch>` component is built from two sub-components,
 * `<Switch.Wrapper>` and `<Switch.Item>`.
 *
 * @component
 */

var index = {
  Wrapper: SwitchWrapper,
  Item: SwitchItem
};

module.exports = index;
