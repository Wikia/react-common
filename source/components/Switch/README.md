```js
class SwitchTest1 extends React.Component {
    constructor() {
        this.state = {
            value: 0,
        };
        this.onChange = this.onChange.bind(this);
    }

    onChange(value) {
        this.setState({value});
    }

    render() {
        return (
            <Switch.Wrapper>
                <Switch.Item onClick={() => this.onChange(0)} active={this.state.value === 0}>Item 0</Switch.Item>
                <Switch.Item onClick={() => this.onChange(1)} active={this.state.value === 1}>Item 1</Switch.Item>
                <Switch.Item onClick={() => this.onChange(2)} active={this.state.value === 2}>Item 2</Switch.Item>
            </Switch.Wrapper>
        );
    }
}

<SwitchTest1 />
```
