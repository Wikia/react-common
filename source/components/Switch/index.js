import SwitchWrapper from '../Switch.Wrapper';
import SwitchItem from '../Switch.Item';

/**
 * `<Switch>` component is built from two sub-components,
 * `<Switch.Wrapper>` and `<Switch.Item>`.
 *
 * @component
 */
export default {
    Wrapper: SwitchWrapper,
    Item: SwitchItem,
};
