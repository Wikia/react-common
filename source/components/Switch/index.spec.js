import { mount } from 'enzyme';
import sinon from 'sinon';
import React from 'react';

import Switch from './index';

test('Switch renders', () => {
    const component = mount(
        <Switch.Wrapper>
            <Switch.Item active>Foo</Switch.Item>
            <Switch.Item>Bar</Switch.Item>
        </Switch.Wrapper>
    );
    expect(component).toMatchSnapshot();
});

test('Switch renders custom class names', () => {
    const component = mount(
        <Switch.Wrapper className="foo">
            <Switch.Item className="foo">Item 1</Switch.Item>
            <Switch.Item className="bar" active>Bar</Switch.Item>
        </Switch.Wrapper>
    );
    expect(component).toMatchSnapshot();
});

test('Switch.Item onClick', () => {
    const mockOnClick = sinon.spy();
    const wrapper = mount(
        <Switch.Item onClick={mockOnClick}>clickable</Switch.Item>
    );

    wrapper.find('button').simulate('click');

    expect(mockOnClick.calledOnce).toBe(true);
});
