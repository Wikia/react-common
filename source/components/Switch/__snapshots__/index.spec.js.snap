// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders 1`] = `
<SwitchWrapper>
  <div
    className="switch-wrapper"
  >
    <SwitchItem
      active={true}
    >
      <button
        className="switch-item switch-item__active"
        type="button"
      >
        Foo
      </button>
    </SwitchItem>
    <SwitchItem
      active={false}
    >
      <button
        className="switch-item"
        type="button"
      >
        Bar
      </button>
    </SwitchItem>
  </div>
</SwitchWrapper>
`;

exports[`Switch renders custom class names 1`] = `
<SwitchWrapper
  className="foo"
>
  <div
    className="switch-wrapper foo"
  >
    <SwitchItem
      active={false}
      className="foo"
    >
      <button
        className="switch-item foo"
        type="button"
      >
        Item 1
      </button>
    </SwitchItem>
    <SwitchItem
      active={true}
      className="bar"
    >
      <button
        className="switch-item bar switch-item__active"
        type="button"
      >
        Bar
      </button>
    </SwitchItem>
  </div>
</SwitchWrapper>
`;
